ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -361,6 +361,99 @@ public class Manager: NSObject, HTTPDownloadDelegate, UIGestureRecognizerDelegat
     }
     return nil
   }
+    
+  // MARK: - Adhoc keyboards
+
+  
+  public func parseKMP(_ folder: URL, complete: @escaping (Int) -> Void) {
+    do {
+      var path = folder
+      path.appendPathComponent("kmp.json")
+      let data = try Data(contentsOf: path, options: .mappedIfSafe)
+      let jsonResult = try JSONSerialization.jsonObject(with: data, options: .mutableLeaves)
+      if let jsonResult = jsonResult as? [String:AnyObject] {
+        if let keyboards = jsonResult["keyboards"] as? [[String:AnyObject]] {
+          for k in keyboards {
+            let name = k["name"] as! String
+            let keyboardID = k["id"] as! String
+            let version = k["version"] as! String
+            let osk = k["oskFont"] as! String
+            let font = k["displayFont"] as! String
+            let displayFont = Font(filename: font)
+            let oskFont = Font(filename: osk)
+
```

Comment:
`oskFont` and `displayFont` are [optional](https://github.com/keymanapp/keyman/wiki/KMP-Metadata-File-(kmp.inf-and-kmp.json)#oskfont) in kmp.json.

Reply:
this has been updated in the latest push